src/attached_probe.cpp

```diff
@@ -66,6 +67,7 @@ libbpf::bpf_prog_type progtype(ProbeType t)
   switch (t) {
       // clang-format off
     case ProbeType::special:    return libbpf::BPF_PROG_TYPE_RAW_TRACEPOINT; break;
+    case ProbeType::benchmark:  return libbpf::BPF_PROG_TYPE_XDP; break;
```

Comment:
why is this probe an XDP type?

Reply:
Unfortunately, `BPF_PROG_TYPE_RAW_TRACEPOINT` and the other probe types here do not support the `.repeat` option [in the kernel](https://github.com/jrife/linux/blob/90b83efa6701656e02c86e7df2cb1765ea602d07/net/bpf/test_run.c#L769).  `BPF_PROG_TYPE_XDP` happens to be one type that does.

Reply:
Ah ok. I still wonder if it’s easier to make this a special probe type and just check for the name when setting the attach type but I won’t block on this detail.

Reply:
I may try that configuration and see how messy the code feels. If it looks gross, I'll just go with the different probe types. Since this is all internal, I guess they could be merged later if need be. Overall, I just want to avoid bunch of special-casing around `BENCH`.